changelog.d/+deprecate-header-filter.removed.md

```diff
@@ -0,0 +1 @@
+Remove http_header_filter
```

Comment:
Add more information on what users should use instead..
"Please use `http_filter` with key `header_filter` instead"